include/admin/class-monero-admin-payments-list.php

```diff
@@ -0,0 +1,276 @@
+
+    }
+
+    function get_columns() {
+        return $columns= array(
+            'col_order_id' => __('Order'),
+            'col_payment_id' => __('Payment ID'),
+            'col_txid' => __('Txid'),
+            'col_height' => __('Height'),
+            'col_amount' => __('Amount'),
+        );
+    }
+
+    public function get_sortable_columns() {
+        return array();
+        return $sortable = array(
+            'col_order_id' => 'col_order_id',
+            'col_payment_id' => 'payment_id',
+            'col_txid' => 'txid',
+            'col_height' => 'height',
+            'col_amount' => 'amount',
+        );
+    }
+
+    function prepare_items() {
+
+        $this->_column_headers = array($this->get_columns(), array(), $this->get_sortable_columns());
+        $current_page = absint($this->get_pagenum());
+
+        $per_page = 25;
+
+        $this->get_items($current_page, $per_page);
+
+    }
+
+    public function no_items() {
+        esc_html_e('No Monero payments found', 'monero_gateway');
+    }
+
+    protected function get_filter_vars() {
+        $type = isset($_GET['type']) ? $_GET['type'] : null;
+        return (object) array(
+            'type' => $type,
+        );
+    }
+
+    protected function get_item_count($type) {
+        global $wpdb;
+        $table_name_1 = $wpdb->prefix.'monero_gateway_quotes';
+        $table_name_2 = $wpdb->prefix.'monero_gateway_quotes_txids';
+        $query_where = ' WHERE 1=1 '.$this->get_clause_type($type);
+        $query = "SELECT COUNT(*) AS count FROM {$table_name_2} t2 LEFT JOIN $table_name_1 t1 ON t2.payment_id = t1.payment_id {$query_where}";
+        $item_count = $wpdb->get_var($query);
+        if(is_null($item_count)) $item_count = 0;
+        return $item_count;
+    }
+
+    protected function get_clause_type($type) {
+        global $wpdb;
+        switch($type) {
+        case 'pending':
+            $query_where = $wpdb->prepare(' AND pending = 1 AND paid = 0 ', array());
+            break;
+        case 'paid':
+            $query_where = $wpdb->prepare(' AND paid = 1 AND confirmed = 0 ', array());
+            break;
+        case 'confirmed':
+            $query_where = $wpdb->prepare(' AND confirmed = 1 ', array());
+            break;
+        case 'expired':
+            $query_where = $wpdb->prepare(' AND paid = 0 AND pending = 0 ', array());
+            break;
+        case 'all':
+        default:
+            $query_where = ' ';
+        }
+        return $query_where;
+    }
+
+    public function get_items($current_page, $per_page) {
+        global $wpdb;
+
+        $this->items = array();
+        $filters = $this->get_filter_vars();
+
+        $table_name_1 = $wpdb->prefix.'monero_gateway_quotes';
+        $table_name_2 = $wpdb->prefix.'monero_gateway_quotes_txids';
+
+        $query_where = ' WHERE 1=1 ';
```

Comment:
?

Reply:
This is a common style of coding `WHERE` clauses when the conditions are optional. It allows you to add ` AND <CONDITION>` to the end of it without worrying about syntax.  See: https://dba.stackexchange.com/a/54966

Reply:
It's a bit hacky and it adds some times to execute that SQL query.

Reply:
From the link `and it has no impact on execution time`. I do not think it has impact on execution time. This method is used 31 times in wordpress core and woocommerce. `grep -Ri '1=1' . | wc -l`